tests/zquantum/core/estimation/estimation_test.py

```diff
@@ -1,55 +1,26 @@
+from functools import partial
+
```

Comment:
(This applies to the tests in general.) You mentioned some of the tests are expected to fail. WDYT about marking these tests with [skip](https://docs.pytest.org/en/latest/how-to/skipping.html) in a single, separate commit? This way we should be able to easily re-enable them when we're ready.

Reply:
So adding sth like:
`@pytest.mark.skip(reason="These tests will fail until CostFunction is migrated to new circuits")`
?

Reply:
Just wondering – is it better to use `skip` or `xfail` in such case?

Reply:
I'd use `skip` because it doesn't actually run the tests, saving some time + it provides a natural place to add a commentary